internal/provider/resource_switch_port_settings.go

```diff
@@ -0,0 +1,962 @@
+	for _, link := range settings.Links {
+		links = append(links, switchPortSettingsLinkModel{
+			Autoneg:  types.BoolPointerValue(link.Autoneg),
+			FEC:      types.StringValue(string(link.Fec)),
+			LinkName: types.StringValue(string(link.LinkName)),
+			LLDP: &switchPortSettingsLinkLLDPModel{
+				ChassisID:         types.StringValue(link.LldpLinkConfig.ChassisId),
+				Enabled:           types.BoolPointerValue(link.LldpLinkConfig.Enabled),
+				LinkDescription:   types.StringValue(link.LldpLinkConfig.LinkDescription),
+				LinkName:          types.StringValue(string(link.LinkName)),
+				ManagementIP:      types.StringValue(link.LldpLinkConfig.ManagementIp),
+				SystemDescription: types.StringValue(link.LldpLinkConfig.SystemDescription),
+				SystemName:        types.StringValue(link.LldpLinkConfig.SystemName),
+			},
+			MTU:   types.Int32Value(int32(*link.Mtu)),
+			Speed: types.StringValue(string(link.Speed)),
+			TxEq: &switchPortSettingsLinkTxEqModel{
+				Main:  types.Int32Value(int32(*link.TxEqConfig.Main)),
+				Post1: types.Int32Value(int32(*link.TxEqConfig.Post1)),
+				Post2: types.Int32Value(int32(*link.TxEqConfig.Post2)),
+				Pre1:  types.Int32Value(int32(*link.TxEqConfig.Pre1)),
+				Pre2:  types.Int32Value(int32(*link.TxEqConfig.Pre2)),
+			},
+		})
+	}
+	model.Links = links
+
+	routesMap := make(map[string][]switchPortSettingsRouteRouteModel)
+	for _, route := range settings.Routes {
+		if _, ok := routesMap[string(route.InterfaceName)]; !ok {
+			routesMap[string(route.InterfaceName)] = make([]switchPortSettingsRouteRouteModel, 0)
+		}
+
+		routesMap[string(route.InterfaceName)] = append(
+			routesMap[string(route.InterfaceName)],
+			switchPortSettingsRouteRouteModel{
+				Dst:         types.StringValue(route.Dst.(string)),
+				GW:          types.StringValue(route.Gw),
+				RIBPriority: types.Int32Value(int32(*route.RibPriority)),
+				VID:         types.Int32Value(int32(*route.VlanId)),
+			},
+		)
+	}
+	routes := make([]switchPortSettingsRouteModel, 0)
+	for linkName, rts := range routesMap {
+		routes = append(routes, switchPortSettingsRouteModel{
+			Routes:   rts,
+			LinkName: types.StringValue(linkName),
+		})
+	}
+	model.Routes = routes
+
+	return model, diags
+}
+
+func toOxideParams(model switchPortSettingsModel) (oxide.NetworkingSwitchPortSettingsCreateParams, diag.Diagnostics) {
+
+	params := oxide.NetworkingSwitchPortSettingsCreateParams{
+		Body: &oxide.SwitchPortSettingsCreate{
+			Name:        oxide.Name(model.Name.ValueString()),
+			Description: model.Description.ValueString(),
+			PortConfig: oxide.SwitchPortConfigCreate{
+				Geometry: oxide.SwitchPortGeometry(model.PortConfig.Geometry.ValueString()),
+			},
+		},
+	}
+
+	addresses := make([]oxide.AddressConfig, 0)
+	for _, address := range model.Addresses {
+		addrs := make([]oxide.Address, 0)
+		for _, addr := range address.Addresses {
+			addrs = append(addrs, oxide.Address{
+				Address:    addr.Address,
```

Comment:
This might fix the error

```suggestion
				Address:    addr.Address.ValueString(),
```

Or perhaps:
```suggestion
				Address:    oxide.IpNet(addr.Address.ValueString()),
```